lib/classroom_api.rb

```diff
@@ -222,79 +261,92 @@ def create_classroom_characteristics(characteristics)
+        result['data'] = response_json['Classrooms']
+      elsif response_json['errorCode'].present?
+        result['errorcode'] = response_json['errorCode']
+        result['error'] = response_json['errorMessage']
+      else 
+        result['errorcode'] = "Unknown error"
+      end
+    rescue StandardError => e
+      result['errorcode'] = "Exception"
+      result['error'] = e.message
     end
     return result
 
   end
 
   def update_all_classroom_contacts
-    result = {'success' => false, 'errorcode' => '', 'error' => '', 'data' => {}}
-    @debug = false
-    classrooms = Room.where(rmtyp_description: "Classroom").where.not(facility_code_heprod: nil)
-    number_of_api_calls_per_minutes = 0
-    redo_loop_number = 1
-    classrooms.each do |room|
-      if number_of_api_calls_per_minutes < 400
-        number_of_api_calls_per_minutes += 1
-      else
-        @log.api_logger.debug "update_all_classroom_contacts, the script sleeps after #{number_of_api_calls_per_minutes} calls"
-        number_of_api_calls_per_minutes = 1
-        sleep(61.seconds)
-      end
-      facility_id = room.facility_code_heprod
-      rmrecnbr = room.rmrecnbr
-      result = get_classroom_contact(ERB::Util.url_encode(facility_id))
-      if result['errorcode'] == "ERR429"
-        @log.api_logger.debug "update_all_classroom_contacts, error: API return: #{result['errorcode']} - #{result['error']} after #{number_of_api_calls_per_minutes} calls"
-        number_of_api_calls_per_minutes = 0
-        sleep(61.seconds)
-        if redo_loop_number > 9
-          @debug = true
-          @log.api_logger.debug "update_all_classroom_contacts, error: API return: #{result['errorcode']} - #{result['error']} after #{number_of_api_calls_per_minutes} calls #{redo_loop_number} times "
-          return @debug
+    begin
+      result = {'success' => false, 'errorcode' => '', 'error' => '', 'data' => {}}
+      @debug = false
+      classrooms = Room.where(rmtyp_description: "Classroom").where.not(facility_code_heprod: nil)
+      number_of_api_calls_per_minutes = 0
+      redo_loop_number = 1
+      classrooms.each do |room|
+        if number_of_api_calls_per_minutes < 400
+          number_of_api_calls_per_minutes += 1
+        else
+          @log.api_logger.debug "update_all_classroom_contacts, the script sleeps after #{number_of_api_calls_per_minutes} calls"
+          number_of_api_calls_per_minutes = 1
+          sleep(61.seconds)
         end
-        redo_loop_number += 1
-        redo
-      elsif result['success']
-        if result['data']['Classrooms'].present?
-          row = result['data']['Classrooms']['Classroom'][0]
-          
-          if RoomContact.find_by(rmrecnbr: rmrecnbr).present?
-            update_classroom_contact(row, rmrecnbr)
+        facility_id = room.facility_code_heprod
+        rmrecnbr = room.rmrecnbr
+        result = get_classroom_contact(ERB::Util.url_encode(facility_id))
+        if result['errorcode'] == "ERR429"
```

Comment:
Consider creating a constant variable for this hardcoded value.

Reply:
done here [d203cb6](https://github.com/lsa-mis/mi_classrooms/pull/326/commits/d203cb6598de3f3f06e79e271c57e0e9e1931b0d)